src/OpenCensus.Exporter.Prometheus/OpenCensus.Exporter.Prometheus.csproj

```diff
@@ -19,6 +19,8 @@
 
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore" Version="2.2.0" />
```

Comment:
Hm...
This and next package dependencies could be a bit problematic.
It means that if I run, client-server app that is not using asp.net core (for example, using grpc for communication), I will still have to take dependency on asp.net core packages.
Adding @SergeyKanzhelev to comment on this.

Reply:
Good point. Similarly if we were to add a gRPC interceptor in this library, users of MVC would have gRPC libraries in their dependency tree

Reply:
I bet you only need a reference on `Microsoft.AspNetCore.Http.Abstractions`. This will minimize the "impact" of this package. 

I think it may be even better to have a separate package that will only implement extension method and middleware "wrapper".

Reply:
I've retooled the implementation to be a separate package that is just the middleware and wrapper, this has some unintended consequences to the internals of the Prometheus Exporter, in that the Implementation namespace has several internal classes that I have made public, or fields I have moved. Specifically, the `ContentType` variable I have moved to the `PrometheusExporter` class, and the `MetricsWriter` class I have made a `public` class.

How does that mesh with the design philosophy?

Reply:
@orthros it's a good question. Basically what minimum API we need to expose that will allow to build host-independent headers access and stream writing. We will face the same problems for zPages so I'm glad you've experimented with this already.

I'd suggest some callback approach for headers access similar to `Inject` and `Extract` used for propagation: https://github.com/census-instrumentation/opencensus-csharp/blob/af31c9580a5d0dbdd7d73ae0444fbf546425e494/src/OpenCensus.Abstractions/Trace/Propagation/ITextFormat.cs#L35-L51 And some basic interface to write content to stream. What you listed sounds about right.